src/engine/shared/datafile.cpp

```diff
@@ -249,6 +249,9 @@ void *CDataFileReader::GetDataImpl(int Index, int Swap)
 {
 	if(!m_pDataFile) { return 0; }
 
+	if(Index < 0 || ((unsigned) Index) >= m_pDataFile->m_Header.m_NumRawData)
```

Comment:
Why the cast to unsigned? NumRawData is int
Same for the next 2 cases.

Reply:
You're right. Went back and forth with the code, I don't see any reason to keep it.